# Sikiric Merjem - Graduation Project - Question 11

In [1]:
import scipy.stats as stats
import pandas as pd
import numpy as np

dataset11 = pd.read_csv("Survey__results.csv")
dataset11.head()

ID Please select your age: Please select your gender:  \
0   1                      24                     Female   
1   2                      19                     Female   
2   3                      19                       Male   
3   4                      18                       Male   
4   5                      20                       Male   

  What is your nationality? What is your current academic year or level?  \
0                   Bosnian                 Senior (fourth year student)   
1                   Bosnian                Freshman (first year student)   
2                   Bosnian                Freshman (first year student)   
3                   Turkish                Freshman (first year student)   
4                   Bosnian                Freshman (first year student)   

  What influenced your choice of field of study in Computer Science?   \
0               Personal interest;Job opportunities;                    
1                                 Personal interest;                    
2                                 Job opportunities;                    
3                                 Personal interest;                    
4                                 Personal interest;                    

       What are you planning to do after graduation?  \
0  Find a job in the industry;Start my own business;   
1           Continue my graduate studies (master’s);   
2                        Find a job in the industry;   
3            Start my own business;I don't know yet;   
4                                  I don't know yet;   

  Please describe “academic career” in your own words. What do you know about academic career? What comes to your mind?  \
0  An academic career is like a journey where peo...                                                                      
1                                  studying for life                                                                      
2                                                  .                                                                      
3     Learning is way more important than a diploma                                                                       
4                                               Good                                                                      

  What do you think are the motivational factors to pursue for an academic career in Computer Science?   \
0  Interest and confidence in research (academic ...                                                      
1  Interest and confidence in research (academic ...                                                      
2                         Flexible work environment;                                                      
3  Job opportunities in universities;Flexible wor...                                                      
4                         Flexible work environment;                                                      

  What do you think are the demotivational factors to pursue academic career in Computer Science?   \
0  Lack of confidence;Lack of information and gui...                                                 
1     Stressful work environment;Lack of confidence;                                                 
2                                                 .;                                                 
3                                Financial concerns;                                                 
4                                Financial concerns;                                                 

  Do you think there is a gender gap in the IT sector in BiH?   \
0                                                Yes             
1                                           Not sure             
2                                           Not sure             
3                                                Yes             
4                                                Yes             

  What are your t

11 - What are your thoughts on the representation of women in your field?

Here we check the relationship between male and female responses for the perception of women representation. We observe the 'Please select your gender:' and 'What are your thoughts on the representation of women in your field:' fields.

Null and Alternative Hypothesis:

Ho:For the population of IT students in BiH, gender and the perception of women representation in the field is not related. 

H1:For the population of IT students in BiH, gender and the perception of women representation in the field is related.

In [2]:
#number of responses by each gender is summarized in the table
dataset11_table=pd.crosstab(dataset11['Please select your gender:'],dataset11['What are your thoughts on the representation of women in your field:'])
print(dataset11_table)

What are your thoughts on the representation of women in your field:  Not sure  \
Please select your gender:                                                       
Female                                                                      10   
Male                                                                        24   

What are your thoughts on the representation of women in your field:  Women are underrepresented  \
Please select your gender:                                                                         
Female                                                                                        32   
Male                                                                                          10   

What are your thoughts on the representation of women in your field:  Women are well-represented  
Please select your gender:                                                                        
Female                                                                  

In [3]:
val=stats.chi2_contingency(dataset11_table)
val

Chi2ContingencyResult(statistic=30.940799653194613, pvalue=1.911132167615841e-07, dof=2, expected_freq=array([[14.7027027 , 18.16216216, 15.13513514],
       [19.2972973 , 23.83783784, 19.86486486]]))

In [5]:
#Calculate degrees of freedom
nb_rows=len(df.iloc[0:2,0])
nb_cols=len(df.iloc[0,0:3])
dof=(nb_rows-1)*(nb_cols-1)
alpha=0.05

#Observed values = oVal11
#Expected values = eVal11
oVal11=dataset11_table.values
eVal11=val[3]

Chi Square formula: \begin{equation}
x^2 = ∑ (O − E)^2 / E  
\end{equation}

In [6]:
from scipy.stats import chi2
chi_square=sum([(o-e)**2./e for o,e in zip(oVal11, eVal11)])
chi_statistics=chi_square[0]+chi_square[1]+chi_square[2]
print('chi value: ', chi_statistics)

chi value:  30.940799653194617


In [7]:
critical_val=chi2.ppf(q=1-alpha,df=dof)
print('critical value: ', critical_val)

critical value:  5.991464547107979


If Chi-Square value (chi_statistics) is greater than 5.99, reject Ho. (meaning that meaning that perception of womens representation in the field is differing from males and females percpective)

In [8]:
if chi_statistics>=critical_val:
    print("Reject Ho: There is a relationship between gender and perception of women's representation in the field")
else:
    print("Accept Ho: There is no relationship between 2 variables.")

Reject Ho: There is a relationship between gender and perception of women's representation in the field


In [9]:
#calculate p-value -alternatively
pVal=1-chi2.cdf(x=chi_statistics,df=dof)
print('p value= ', pVal)
if pVal<=alpha:
    print("Reject Ho: There is a relationship between gender and perception of women's representation in the field")
else:
    print("Accept Ho: There is no relationship between 2 variables.")

p value=  1.911132168075369e-07
Reject Ho: There is a relationship between gender and perception of women's representation in the field


# Effect Size


Effect size is a measure of the power of the association between the two features. We use Cramer's V value to see the effect size.

In [10]:
    data = np.array([[1, .1, .3, .5],
       [2, .07, .21, .35],
       [3, .06, .17, .29],
       [4, .05,.15,.25],
       [5, .04, .13, .22]])
    sizes = pd.DataFrame(data, columns=['Degrees of Freedom', 'Small Effect', 'Medium Effect', 'Large Effect']) 
    
    # getting the chi sq. stat
    chi2 = val[0]
    # calculating the total number of observations
    n = dataset11_table.sum().sum()
    # getting the degrees of freedom
    dof = min(dataset11_table.shape)-1
    # calculating cramer's v
    v = np.sqrt(chi2/(n*dof))
    # printing results
    print(f'V = {v}')
    print(f'Cramer\'s V Degrees of Freedom = {dof}')
    print(f'\nEffect Size Thresholds\n{sizes}\n')

V = 0.5279639597752607
Cramer's V Degrees of Freedom = 1

Effect Size Thresholds
   Degrees of Freedom  Small Effect  Medium Effect  Large Effect
0                 1.0          0.10           0.30          0.50
1                 2.0          0.07           0.21          0.35
2                 3.0          0.06           0.17          0.29
3                 4.0          0.05           0.15          0.25
4                 5.0          0.04           0.13          0.22



The Cramer’s V result indicates that gender has a large, statistically significant effect on the perception of women representation in the field.